model/config.go

```diff
@@ -227,6 +229,7 @@ type ServiceSettings struct {
 	MaximumLoginAttempts                              *int
 	GoroutineHealthThreshold                          *int
 	GoogleDeveloperKey                                *string
+	ActionCookieSecret                                *string
```

Comment:
Can you help describe the purpose of this setting - is it for security reasons? Is this absolutely necessary for admins to configure?

Reply:
If we add this setting, we should 
- add its default value to config/default.json
- add if the default value is used or not to app/diagnostics.go

Reply:
It is the encryption key used to encrypt the cookies (plugin's Integration Context), specifically and hide them from the client. I does auto-default to a random value, but in a clustered config it should be set the same across all servers.

I'm 2/5 that tracking the default/override usage for this is not useful, would recommend against it. It should be unique for every cluster.

Reply:
> I does auto-default to a random value

That means if someone upgrades to a release this feature ships in, and attempts to use PostActions in an ephemeral message, it will work out-of-the-box, and not require admin to configure/set the encryption key?

Reply:
As far as I understand, correct, https://github.com/mattermost/mattermost-server/pull/10258/files#diff-2279f86634da3f787f279961af76713aR320 takes care of it.

Reply:
Okay. One more question - any objection for adding this setting in System Console UI, say in System Console > Integrations > Custom Integrations?

It would give the admin an ability to regenerate a new secret easily, like for public links (see below)
![image](https://user-images.githubusercontent.com/13119842/52522978-11462800-2c5a-11e9-8e3c-788ec69ad0c4.png)

Reply:
Thank you this is exactly the feedback I needed (need to add this to console and docs)! Just need to confirm that this is the right thing in a clustered environment to have the value shared to all servers.

Reply:
The server has a `asymmetricSigningKey` which is a `ecdsa.PrivateKey`(https://github.com/mattermost/mattermost-server/blob/master/app/server.go#L100). Maybe this can be used to encrypt the cookie? I only know about the existence, Christopher will have more insights in this.

Reply:
@hanzei @crspeller asymmetricSigningKey does not seem to be the right thing to use. 0/5 should stick with a separate setting. Let me know what you think.

Also, @crspeller can you please confirm that the current implementation is correct for the clustered use-case, and that I would just need to add this key to admin panel as Jason suggested?

Reply:
You're right, `ecdsa.PrivateKey` just exposes a `Sign()` method. We could still do something fancy like generating a ephemeral symmetry key, signing it and sending the key and the signature to the client. We would later check the returned signature and then encrypt with the provided key. 0/5 that this is the right way.

Technically elliptic curve public/private keys are able to encrypt data. `crypto/ecdsa` just doens't support this. But I don't know a library that can do this.

There are two reasons why I'm persistent on this and I think this should not be a system setting:
- It adds jet another system setting
- It exposes something security related to the system admin even tough the admin should not care (and can only screw thing up).
- A sign/encrypt function seams to be useful in general.

Reply:
I agree that we should try hard not to expose the setting at all, it is just an internally-used 32 bytes that should be cryptographically random, and shared between all servers in a cluster. I am still unsure that piggybacking on PKI is a good idea here.